File: ativAula4.js
// Programa de contagem regressiva
let tempo = 10; // inicializa a variável tempo

while (tempo > 0) {
    // enquanto tempo for maior que 0
    console.log(tempo); // exibe o valor de tempo
    tempo--; // decrementa o valor de tempo
}

// depois do loop, exibe a mensagem de Ano Novo
console.log("Feliz Ano Novo!");


// Soma dos números pares de 1 a 20
let sominha = 0; // inicializa a variável para a soma

for (let numero = 2; numero <= 20; numero += 2) {
    // itera sobre os números pares de 2 a 20
    sominha += numero; // adiciona o número à soma
}

console.log(`A soma dos números pares de 1 a 20 é: ${sominha}`);


// Tabuada do Número 7
const numeroTabuada = 7; // define o número para a tabuada

for (let contador = 1; contador <= 10; contador++) {
    // itera de 1 a 10
    console.log(`${numeroTabuada} x ${contador} = ${numeroTabuada * contador}`); // exibe o resultado da multiplicação
}


// Contagem de Vogais em uma Palavra
const vogais = ["a", "e", "i", "o", "u"]; // lista de vogais
const palavra = "programador"; // palavra para verificação

let totalVogais = 0; // contador de vogais

for (let caractere of palavra) {
    // itera sobre cada caractere da palavra
    if (vogais.includes(caractere)) {
        // verifica se o caractere é uma vogal
        totalVogais++; // incrementa o contador
    }
}

console.log(`A palavra "${palavra}" contém ${totalVogais} vogais.`); // exibe o resultado


// Contagem de Números Ímpares de 1 a 20
let contador = 1; // inicia o contador

while (contador <= 20) {
    // enquanto o contador for menor ou igual a 20
    if (contador % 2 === 1) {
        // verifica se o número é ímpar
        console.log(`Número ímpar: ${contador}`); // exibe o número ímpar
    }
    contador++; // incrementa o contador
}


// Soma dos Dígitos de um Número Inteiro
const numero = 5678; // número para calcular a soma dos dígitos

let soma = 0; // inicializa a variável soma

const numeroComoString = numero.toString(); // converte o número para string

for (let digito of numeroComoString) {
    // itera sobre cada dígito
    soma += parseInt(digito); // converte o dígito para número e soma
}

console.log(`A soma dos dígitos do número ${numero} é: ${soma}`); // exibe o resultado 


// // Propriedades de um Objeto
let pessoa = { // cria um objeto com detalhes de uma pessoa
    nome: "Carlos Ferreira",  // nome da pessoa
    idade: 35,  // idade
    endereco: {  // endereço completo
        rua: "Rua dos Girassóis",  // nome da rua
        numero: 789,  // número da residência
        cidade: "Vale do Sol",  // nome da cidade
    }
};

// exibe as informações formatadas
console.log(`Nome: ${pessoa.nome}`);
console.log(`Idade: ${pessoa.idade}`);
console.log(`Endereço: ${pessoa.endereco.rua}, ${pessoa.endereco.numero}, ${pessoa.endereco.cidade}`);

File: ativAula4.py
# Programa de contagem regressiva
time = 10  # define a variável time

while time >= 1:
    # enquanto time for maior ou igual a 1
    print(time)  # exibe o valor de time
    time -= 1  # reduz o valor de time

# após o loop, exibe a mensagem de Ano Novo
print("Feliz Ano Novo!")


# Soma dos números pares de 1 a 20
soma = 0  # Inicializa a variável para armazenar a soma

for num in range(1, 21): # Itera sobre os números de 1 a 20
    if num % 2 == 0:  # verifica se o número é par
        soma += num  # adiciona o número à soma

print(f'A soma dos números pares de 1 a 20 é: {soma}')


# Tabuada do número 7
numero_tabuada = 7  # define o número para a tabuada

for i in range(1, 11):
    # itera sobre os números de 1 até 10
    print(f'{numero_tabuada} x {i} = {numero_tabuada * i}')  # calcula e exibe o produto do número da tabuada com i


# Contagem de vogais em uma palavra
vogais = ["a", "e", "i", "o", "u"]  # lista de vogais
palavra = "programador"  # palavra para analisar

contagem = 0  # inicializa o contador de vogais

for letra in palavra:
    # verifica cada letra da palavra
    if letra in vogais:
        # se a letra estiver na lista de vogais, incrementa o contador
        contagem += 1

print(f'A palavra "{palavra}" tem {contagem} vogais.')  # mostra o número total de vogais na palavra


# Contagem de números ímpares de 1 a 20
contagem = 1  # inicia a variável de contagem

while contagem <= 20:
    # enquanto a contagem for menor ou igual a 20
    if contagem % 2 != 0:
        # se a contagem for um número ímpar
        print(f'Número ímpar: {contagem}')  # exibe o número ímpar
    contagem += 1  # incrementa a contagem


# Soma dos dígitos de um número inteiro
numero = 1234  # número inteiro para calcular a soma dos dígitos

soma = 0  # inicializa a variável para armazenar a soma dos dígitos


# Converte o número para uma string para iterar sobre cada caractere
numero_string = str(numero)

for caractere in numero_string:
    # itera sobre cada caractere na representação em string do número
    soma += int(caractere)  # converte o caractere para inteiro e adiciona à soma

print(f'A soma dos dígitos do número {numero} é: {soma}')  # exibe o resultado


# Propriedades de um Objeto
pessoa = { # define um dicionário pessoa com atributos
    "nome": "Alice Souza",  # nome da pessoa
    "idade": 25, # idade
    "endereco": { # endereco
        "rua": "Avenida das Flores", # nome da rua
        "numero": 456, # numero da casa
        "cidade": "Jardim das Estrelas", # cidade
    },
}

# Formatação manual para exibir 
print(f"Nome: {pessoa['nome']}") 
print(f"Idade: {pessoa['idade']}")
print(f"Endereço: {pessoa['endereco']['rua']}, {pessoa['endereco']['numero']}, {pessoa['endereco']['cidade']}")